### PR TITLE
Feature work: Azure functions integration test framework

### PR DESCRIPTION
**Pull request by tippmar-nr**
* Adds a testing framework and a very basic implementation of an Azure function test app and corresponding integration tests. 
* Updates the `all_solutions` workflow to install required tools and include the `AzureFunction` workspace in the integration test matrix. 
* Updates the profiler to exclude `FUNC.EXE` (the local Azure test tool) from instrumentation. Note that I had to update a number of profiler unit tests also, since I'm now referencing `_systemCalls` in areas that tests previously didn't expect.

### CODE REVIEW

File: .github/workflows/all_solutions.yml [hunk 1]
```diff
@@ -279,6 +280,7 @@ jobs:
       # Make this variable true to enable extra data-gathering and logging to help troubleshoot test failures, at the cost of additional time and resources
       enhanced_logging: false
       NR_DOTNET_TEST_SAVE_WORKING_DIRECTORY: 1
+      azure_func_exe_path: C:\ProgramData\chocolatey\lib\azure-functions-core-tools\tools\func.exe
```
Comment (chynesNR) [inline .github/workflows/all_solutions.yml:283]: Is there a variable we can leverage so we don't have to list the full path? I could see it potentially changing with e.g. a new version of Chocolatey

**Review by chynesNR — APPROVED** [review]
Looks good!

File: src/Agent/NewRelic/Profiler/Configuration/Configuration.h [hunk 2]
```diff
@@ -630,25 +632,34 @@ namespace NewRelic { namespace Profiler { namespace Configuration {
...
+             
+            bool isAzureFunctionsNetHost = Strings::EndsWith(processPath, _X("FUNCTIONSNETHOST.EXE"));
             if (isAzureFunctionsNetHost)
             {
                 LogInfo(L"FunctionNetHost.exe is a valid Azure function command. This process will be instrumented.");
                 return 1;
             }
 
+            // func.exe is the local test tool and should not be instrumented
+            bool isFuncExe = Strings::EndsWith(processPath, _X("FUNC.EXE"));
```
Comment (nrcventura) [inline src/Agent/NewRelic/Profiler/Configuration/Configuration.h:656]: Do we think this check is specific enough?

File: tests/Agent/IntegrationTests/Applications/AzureFunctionApplication/.gitignore
Comment (nrcventura) [inline tests/Agent/IntegrationTests/Applications/AzureFunctionApplication/.gitignore:1]: Is this file necessary?

File: tests/Agent/IntegrationTests/IntegrationTests/AzureFunction/AzureFunctionTests.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                new() {metricName = "WebTransactionTotalTime/AzureFunction/Function1", callCount = 1},
+                new() {metricName = "Apdex/AzureFunction/Function1", callCount = 1},
+            };
+
+
+            var metrics = _fixture.AgentLog.GetMetrics().ToList();
+            Assertions.MetricsExist(expectedMetrics, metrics);
+
+            //var transactionSample = _fixture.AgentLog.TryGetTransactionSample("WebTransaction/AzureFunction/Function1");
+            //Assert.NotNull(transactionSample);
```
Comment (nrcventura) [inline tests/Agent/IntegrationTests/IntegrationTests/AzureFunction/AzureFunctionTests.cs:70]: Is this a todo for a future test?

File: src/Agent/NewRelic/Profiler/Configuration/Configuration.h (hunk 2 above)
Comment (tippmar-nr) [reply src/Agent/NewRelic/Profiler/Configuration/Configuration.h:656]: I think so, it'll only be hit if the `FUNCTIONS_WORKER_RUNTIME` environment variable is present, so we know at this point that we're in Azure function mode.

File: tests/Agent/IntegrationTests/Applications/AzureFunctionApplication/.gitignore
Comment (tippmar-nr) [reply tests/Agent/IntegrationTests/Applications/AzureFunctionApplication/.gitignore:1]: It was automatically added when I had VS create the azure function app. Maybe not necessary but doesn't hurt anything.

File: tests/Agent/IntegrationTests/IntegrationTests/AzureFunction/AzureFunctionTests.cs (hunk 3 above)
Comment (tippmar-nr) [reply tests/Agent/IntegrationTests/IntegrationTests/AzureFunction/AzureFunctionTests.cs:70]: Kind of. Until I (or someone) figures out the interaction with aspnetcore instrumentation, the sampled transaction won't be the function transaction. But the integration test implementation isn't complete at this point either - I just wanted to get something working.

File: .github/workflows/all_solutions.yml (hunk 1 above)
Comment (tippmar-nr) [reply .github/workflows/all_solutions.yml:283]: I'd much rather just rely on the azure tools being in the path but that apparently doesn't happen with chocolatey installs. I don't see any obvious way to determine the base path for chocolatey, so I think we'll just have to live with this for now and deal with any breakage that happens in the future.

Comment (tippmar-nr) [reply .github/workflows/all_solutions.yml:283]: Actually it looks like there's a `ChocolateyInstall` environment variable I can leverage. Change coming soon..
